### ISSUE TITLE
on Windows, Limiter() constructor hangs after gevent monkeypatch

### ISSUE THREAD

**Issue by gonewest818** [vutran1710/PyrateLimiter#167]
As the subject says, specifically on Windows this hangs indefinitely:

```
import gevent.monkey
gevent.monkey.patch_all()

from pyrate_limiter import Duration, Rate, Limiter

rate = Rate(5, Duration.SECOND * 2)
limiter = Limiter(rate)
```

Reproduced on Windows 11, using pyenv to install python 3.10, 3.11 and 3.12.

**Comment by vutran1710** [comment]
Were you trying to unit-test the limiter or something?
What version were you using?
And Limiter only accepts a list of rates, not a single rate like in your example.

**Comment by gonewest818** [comment]
No, this is the smallest repro of what I'm seeing in a larger system. I narrowed it down to literally the QuickStart example on your README except with the monkeypatching inserted first...
